framework/src/org/checkerframework/common/basetype/messages.properties

```diff
@@ -18,6 +18,8 @@ bound.type.incompatible=incompatible bounds in %s\ntype: %s\nupper bound: %s\nlo
 monotonic.type.incompatible=cannot assign %s to %s (monotonic type).\nfull type found: %s
 type.invalid=invalid type: annotations %s in type "%s"
 type.invalid.conflicting.annos=invalid type: conflicting annotations %s in type "%s"
+type.invalid.too.many.annotations=invalid type: too many annotations %s in type "%s"
+type.invalid.too.few.annotations=invalid type: missing annotations %s in type "%s"
```

Comment:
At the moment the given annotations are output. So this would be clearer if either the message is changed or the provided value is the top annotations from the missing hierarchies.